src/lib.rs

```diff
@@ -47,47 +47,43 @@ impl Spinner {
     ///
     /// let sp = Spinner::new(Spinners::Dots, String::new());
     /// ```
-    pub fn new(spinner: Spinners, message: String) -> Self {
+    #[must_use] 
+    pub fn new(spinner: Spinners, message: String) -> Self  {
         Self::new_inner(spinner, message, None, None)
     }
 
-    /// Create a new spinner that logs the time since it was created
-    pub fn with_timer(spinner: Spinners, message: String) -> Self {
-        Self::new_inner(spinner, message, Some(Instant::now()), None)
-    }
-
-    /// Creates a new spinner along with a message with a specified output stream
+     /// Create a new colored spinner along with a message
     ///
     /// # Examples
     ///
     /// Basic Usage:
     ///
     /// ```
-    /// use spinners::{Spinner, Spinners, Stream};
-    /// 
-    /// let sp = Spinner::with_stream(Spinners::Dots, String::new(), Stream::Stderr);
-    /// ```
-    pub fn with_stream(spinner: Spinners, message: String, stream: Stream) -> Self {
-        Self::new_inner(spinner, message, None, Some(stream))
-    }
-
-    /// Creates a new spinner that logs the time since it was created with a specified output stream
+    /// use spinners::{Spinner, Spinners, Color};
     ///
-    /// # Examples
+    /// let sp = Spinner::new_with_color(Spinners::Dots, "Loading things into memory...".into(), Color::Blue);
+    /// ```
     ///
-    /// Basic Usage:
+    /// No Message:
     ///
     /// ```
-    /// use spinners::{Spinner, Spinners, Stream};
-    /// 
-    /// let sp = Spinner::with_timer_and_stream(Spinners::Dots, String::new(), Stream::Stderr);
+    /// use spinners::{Spinner, Spinners};
+    ///
+    /// let sp = Spinner::new_with_color(Spinners::Dots, String::new(), None);
     /// ```
-    pub fn with_timer_and_stream(spinner: Spinners, message: String, stream: Stream) -> Self {
```

Comment:
the issue I'm having here is the cardinality of option, someone might want to have a special stream AND a color.

While stream should be set everywhere, I think "color" is one special use-case of text customization and should be exposed like a mapping function you can optionally set + some documentation on spinners on how to customize text color don't you think?